blockchain/blockdao/indexbuilder.go

```diff
@@ -91,7 +91,7 @@ func (ib *IndexBuilder) Indexer() blockindex.Indexer {
 
 // HandleBlock handles the block and create the indices for the actions and receipts in it
 func (ib *IndexBuilder) HandleBlock(blk *block.Block) error {
-	ib.pendingBlks <- blk
+	ib.pendingBlks[blk.Height()] = blk
```

Comment:
define a func addBlockToBuffer()
add incoming block only if its height > current tip, and it does not exist in the map yet